code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -177,25 +210,95 @@
 	if(AM.pulledby)
 		AM.pulledby.stop_pulling()
 
-	return 1
+	return TRUE
 
 /obj/structure/closet/proc/close(mob/living/user)
 	if(!opened || !can_close(user))
-		return 0
+		return FALSE
 	take_contents()
 	playsound(loc, close_sound, 15, 1, -3)
 	climb_time = initial(climb_time)
 	opened = 0
 	density = TRUE
 	update_icon()
-	return 1
+	return TRUE
 
 /obj/structure/closet/proc/toggle(mob/living/user)
 	if(opened)
 		return close(user)
 	else
 		return open(user)
 
+/obj/structure/closet/proc/bust_open()
+	welded = FALSE //applies to all lockers
+	locked = FALSE //applies to critter crates and secure lockers only
+	broken = TRUE //applies to secure lockers only
+	open()
+
+/obj/structure/closet/proc/handle_lock_addition(mob/user, obj/item/electronics/airlock/E)
+	add_fingerprint(user)
+	if(lock_in_use)
+		to_chat(user, "<span class='notice'>Wait for work on [src] to be done first!</span>")
+		return
+	if(secure)
+		to_chat(user, "<span class='notice'>This locker already has a lock!</span>")
+		return
+	if(broken)
+		to_chat(user, "<span class='notice'><b>Unscrew</b> the broken lock first!</span>")
+		return
+	if(!istype(E))
+		return
+	user.visible_message("<span class='notice'>[user] begins installing a lock on [src]...</span>","<span class='notice'>You begin installing a lock on [src]...</span>")
+	lock_in_use = TRUE
+	playsound(loc, 'sound/items/screwdriver.ogg', 50, 1)
+	if(!do_after(user, 200, target = src))
+		lock_in_use = FALSE
+		return
+	if(!user.drop_item())
+		to_chat(user, "<span class='notice'>[E] is stuck to you!</span>")
+		lock_in_use = FALSE
+		return
+	lock_in_use = FALSE
+	to_chat(user, "<span class='notice'>You finish the lock on [src]!</span>")
+	E.forceMove(src)
+	lockerelectronics = E
+	req_access = E.accesses
+	secure = TRUE
+	update_icon()
+	return TRUE
+
+/obj/structure/closet/proc/handle_lock_removal(mob/user, obj/item/screwdriver/S)
+	if(lock_in_use)
+		to_chat(user, "<span class='notice'>Wait for work on [src] to be done first!</span>")
+		return
+	if(locked)
+		to_chat(user, "<span class='notice'>Unlock it first!</span>")
+		return
+	if(!secure)
+		to_chat(user, "<span class='notice'>[src] doesn't have a lock that you can remove!</span>")
+		return
+	if(!istype(S))
+		return
+	var/brokenword = broken ? "broken" : null
+	user.visible_message("<span class='notice'>You begin removing the [brokenword] lock on [src]...</span>", "<span class='notice'>[user] begins removing the [brokenword] lock on [src]...</span>")

```

Comment:
`You begin removing the  lock on...`. This will have a double space between `the` and `lock` for a non-broken lock.